mmocr/datasets/pipelines/transforms.py

```diff
@@ -967,3 +967,50 @@ def generate_crop_target(self, image, all_polys, pad_h, pad_w):
         h_axis = np.where(h_array == 0)[0]
         w_axis = np.where(w_array == 0)[0]
         return h_axis, w_axis
+
+
+@PIPELINES.register_module()
+class PyramidRescale:
+    """Resize the image to the base shape, downsample it with gaussian pyramid,
+    and rescale it back to original size.
+
+    Adapted from https://github.com/FangShancheng/ABINet.
+
+    Args:
+        factor (int): The decay factor from base size, or the number of
+            downsampling operations from the base layer.
+        base_shape (tuple(int)): The shape of the base layer of the pyramid.
+        randomize_factor (bool): If True, the final factor would be a random
+            integer in [0, factor].
+
+    :Required Keys:
+        - | ``img`` (ndarray): The input image.
+
+    :Affected Keys:
+        :Modified:
+            - | ``img`` (ndarray): The modified image.
+    """
+
+    def __init__(self, factor=4, base_shape=(128, 512), randomize_factor=True):
+        self.factor = factor if not randomize_factor else np.random.randint(
+            0, factor + 1)
+        self.base_w, self.base_h = base_shape
```

Comment:
assert isinstance(base_shape, list) or isinstance(base_shape, tuple) 
assert len(base_shape) == 2
or if base_shape is int, change base_shape = [base_shape, base_shape]